resolver/resolver.cc

```diff
@@ -370,8 +371,22 @@ class ResolveConstantsWalk {
     }
 
     static void stubForRbiGeneration(core::MutableContext ctx, const vector<ParentPackageStub> &parentPackageStubs,
-                                     const Nesting *scope, ast::ConstantLit *out) {
+                                     const Nesting *scope, ast::ConstantLit *out, bool possibleGenericType) {
         if (out->symbol.exists()) {
+            // In single-package RBI generation mode, we might have already
+            // resolved a class that we later identified as generic, and
+            // we need to attempt to fix that up.
+            if (!out->symbol.isClassOrModule()) {
+                return;
+            }
+            if (!possibleGenericType) {
+                return;
+            }
+            auto singletonClass = out->symbol.asClassOrModuleRef().data(ctx)->singletonClass(ctx);
+            auto &mixins = singletonClass.data(ctx)->mixins();
+            if (absl::c_find(mixins, core::Symbols::T_Generic()) == mixins.end()) {
+                mixins.emplace_back(core::Symbols::T_Generic());
+            }
```

Comment:
Is it worth pulling this out into a separate function that we can call here and on line ~435?